apps/andi/lib/andi/input_schemas/datasets.ex

```diff
@@ -73,6 +74,57 @@ defmodule Andi.InputSchemas.Datasets do
     Repo.insert_or_update(changeset)
   end
 
+  def save_form_changeset(dataset_id, form_changeset) do
+    form_changes = form_changes_from_changeset(form_changeset)
+    update_from_form(dataset_id, form_changes)
+  end
+
+  # TODO refactor
+  def update_from_form(dataset_id, form_changes) do
+    existing_dataset = get(dataset_id)
+    changeset = InputConverter.andi_dataset_to_full_ui_changeset(existing_dataset)
+
+    technical_changes =
+      changeset
+      |> Changeset.get_change(:technical)
+      |> Changeset.apply_changes()
+      |> StructTools.to_map()
+      |> Map.merge(form_changes)
+
+    business_changes =
+      changeset
+      |> Changeset.get_change(:business)
+      |> Changeset.apply_changes()
+      |> StructTools.to_map()
+      |> Map.merge(form_changes)
+
+    new_changes = %{technical: technical_changes, business: business_changes, id: dataset_id} |> StructTools.to_map()
+
+    Dataset.changeset_for_draft(existing_dataset, new_changes)
+    |> save()
+  end
+
+  defp form_changes_from_changeset(%{changes: %{schema: schema}} = form_changeset) do
```

Comment:
I like this pattern. Is there a reason to not make this more dynamic though? As in, updating StructTools to deal with a list? Maybe I'm missing something?